catalog/MDCCatalog/MDCCatalogTileView.swift

```diff
@@ -33,16 +33,36 @@ class MDCCatalogTileView: UIView {
   private lazy var imageView = UIImageView()
   private let imageCache = NSCache<AnyObject, UIImage>()
 
+  deinit {
+    NotificationCenter.default.removeObserver(self,
+                                              name: AppTheme.didChangeGlobalThemeNotificationName,
+                                              object: nil)
+  }
+
   override init(frame: CGRect) {
     super.init(frame: frame)
+
     self.backgroundColor = UIColor.clear
     self.addSubview(imageView)
+
+    NotificationCenter.default.addObserver(
```

Comment:
In this case we need the observer event to be able to blow away our cache. Moving the observation to the view's lifecycle would require that we detect when the previous color scheme differs from the one we're presently using. This may be a reasonable separate change to make, but we can pursue that in a follow-up improvement (a very low-priority one).